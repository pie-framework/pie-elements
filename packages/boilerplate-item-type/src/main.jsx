import React from 'react';
import PropTypes from 'prop-types';
import { PreviewPrompt } from '@pie-element/pie-utils/render-ui';
import UTILS from '@pie-element/pie-utils';

const {test, callTest} = UTILS;

callTest('player');
export default class Main extends React.Component {
  static propTypes = {
    model: PropTypes.object,
    session: PropTypes.object,
    onSessionChange: PropTypes.func.isRequired,
  };

  static defaultProps = {};

  render() {
    const { model } = this.props;

    return (
      <div>
        {model.prompt && (
          <div>
            <p>{test}</p>
            <p>Boilerplate Item</p>
            <PreviewPrompt prompt={model.prompt}/>
          </div>
        )}
      </div>
    );
  }
}
