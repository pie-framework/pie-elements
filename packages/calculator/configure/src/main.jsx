import React from 'react';
import PropTypes from 'prop-types';
import { layout, TwoChoice } from '@pie-lib/config-ui';
import { CalculatorLayout } from '@pie-element/calculator';
import Typography from '@material-ui/core/Typography';

const MainTypes = {
  model: PropTypes.object.isRequired,
  onChange: PropTypes.func.isRequired,
};

export class Main extends React.Component {
  static propTypes = MainTypes;

  onModeChange = (mode) => {
    const { model, onChange } = this.props;
    const update = { ...model, mode };

    onChange(update);
  };

  render() {
    const { model } = this.props;

    return (
      <layout.ConfigLayout hideSettings={true} settings={null}>
        <TwoChoice
          header="Choose a Calculator"
          value={model.mode}
          onChange={this.onModeChange}
          one={{ label: 'Basic', value: 'basic' }}
          two={{ label: 'Scientific', value: 'scientific' }}
        />
        <br />

        <CalculatorLayout mode={model.mode} onClose={() => ({})} />
        <br />

        <Typography style={{ paddingTop: '4px' }}>
          Please note that the calculators are tools for students and do not record answers.
        </Typography>
      </layout.ConfigLayout>
    );
  }
}

export default class Stateful extends React.Component {
  static propTypes = MainTypes;

  constructor(props) {
    super(props);
    this.state = { model: props.model };
  }

  UNSAFE_componentWillReceiveProps(props) {
    this.setState({ model: props.model });
  }

  onChange = (model) => {
    this.setState({ model }, () => {
      this.props.onChange(this.state.model);
    });
  };

  render() {
    return <Main model={this.state.model} onChange={this.onChange} />;
  }
}
