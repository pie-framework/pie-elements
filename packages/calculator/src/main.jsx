import React from 'react';
import PropTypes from 'prop-types';
import DraggableCalculator from './draggable-calculator';
import CalculatorIcon from './calculator-icon';
import { withStyles } from '@material-ui/core/styles';
import classNames from 'classnames';

const styles = {
  icon: {
    transition: 'fill 200ms',
    '&:hover': {
      fill: 'black',
    },
    cursor: 'pointer',
    verticalAlign: 'middle',
    fill: 'grey',
  },
  active: {
    fill: 'black',
  },
};

class Main extends React.Component {
  static propTypes = {
    model: PropTypes.object.isRequired,
    classes: PropTypes.object.isRequired,
  };
  constructor(props) {
    super(props);
    this.state = { show: false };
  }

  onToggleShow = () => this.setState({ show: !this.state.show });
  onClickClose = () => this.setState({ show: false });

  render() {
    const { show } = this.state;
    const { mode } = this.props.model;
    const { classes } = this.props;
    return (
      <div>
        <CalculatorIcon
          className={show ? classNames(classes.icon, classes.active) : classes.icon}
          onClick={() => this.onToggleShow()}
        />
        <DraggableCalculator mode={mode} show={show} onClose={this.onClickClose} />
      </div>
    );
  }
}

export default withStyles(styles)(Main);
