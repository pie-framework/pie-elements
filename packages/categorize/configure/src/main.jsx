import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import classNames from 'classnames';
import Design from './design';

export class Main extends React.Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    className: PropTypes.string,
    model: PropTypes.object.isRequired,
    onModelChanged: PropTypes.func.isRequired,
    imageSupport: PropTypes.object
  };

  static defaultProps = {};

  render() {
    const {
      classes,
      className,
      model,
      onModelChanged,
      configuration,
      onConfigurationChanged
    } = this.props;

    return (
      <div className={classNames(classes.main, className)}>
        <Design
          imageSupport={this.props.imageSupport}
          title="Design"
          model={model}
          configuration={configuration}
          onChange={onModelChanged}
          onConfigurationChanged={onConfigurationChanged}
        />
      </div>
    );
  }
}
const styles = theme => ({
  main: {}
});

export default withStyles(styles)(Main);
