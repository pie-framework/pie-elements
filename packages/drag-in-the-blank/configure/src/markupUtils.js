import escape from 'lodash/escape';
import isUndefined from 'lodash/isUndefined';

const replaceHtmlRegex = /<[^>]*>?/gm;

const createElementFromHTML = htmlString => {
  const div = document.createElement('div');

  div.innerHTML = htmlString.trim();

  return div;
};

export const processMarkup = markup => {
  const newMarkup = markup.replace(/(\\t)|(\\n)/g, '').replace(/\\"/g, '"').replace(/\\\//g, '/');
  const slateMarkup = createElementFromHTML(newMarkup);
  const choices = [];
  let index = 0;

  slateMarkup.querySelectorAll('[data-type="drag_in_the_blank"]').forEach(s => {
    let value = s.dataset.value && s.dataset.value.replace(/&nbsp;/g, ' ').trim();

    if (!value) {
      value = '';
    }

    choices.push({
      value,
      id: s.dataset.id
    });

    s.replaceWith(`{{${index++}}}`);
  });

  return {
    markup: slateMarkup.innerHTML,
    choices: choices,
    correctResponse: choices.reduce((obj, c, index) => {
      obj[index] = !isUndefined(c.id) && c.id || '';

      return obj;
    }, {})
  };
};

const REGEX = /\{\{(\d+)\}\}/g;

export const createSlateMarkup = (markup, choices, correctResponse) => {
  const newMarkup = markup.replace(/(\\t)|(\\n)/g, '').replace(/\\"/g, '"').replace(/\\\//g, '/');
  let index = 0;

  return newMarkup.replace(REGEX, (match, g) => {
    const correctId = correctResponse[g];
    let correctChoice = choices.find(c => c.id === correctId);

    if (!correctChoice || !correctChoice.value) {
      correctChoice = {
        id: '',
        value: ''
      };
    }

    return `<span data-type="drag_in_the_blank" data-index="${index++}" data-id="${correctChoice.id}" data-value="${escape(correctChoice.value)}"></span>`;
  });
};

export const choiceIsEmpty = choice => {
  if (choice) {
    const { value = '' } = choice;

    return value.trim() === '' || value.replace(replaceHtmlRegex, '') === '';
  }

  return false;
};
