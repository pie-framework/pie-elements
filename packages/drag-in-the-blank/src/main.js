import React from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';
import CorrectAnswerToggle from '@pie-lib/correct-answer-toggle';
import { DragInTheBlank } from '@pie-lib/mask-markup';
import { withDragContext } from '@pie-lib/drag';
import { color, Collapsible, hasText } from '@pie-lib/render-ui';
import { renderMath } from '@pie-lib/math-rendering';
import { withStyles } from '@material-ui/core/styles';

const DraggableDragInTheBlank = withDragContext(DragInTheBlank);

export class Main extends React.Component {
  static propTypes = {
    classes: PropTypes.object,
    model: PropTypes.object,
    value: PropTypes.object,
    feedback: PropTypes.object,
    onChange: PropTypes.func
  };

  static defaultProps = {
    value: {}
  };

  state = {
    showCorrectAnswer: false
  };

  componentDidUpdate() {
    // eslint-disable-next-line
    const domNode = ReactDOM.findDOMNode(this);

    renderMath(domNode);
  }

  toggleShowCorrect = () => {
    this.setState({ showCorrectAnswer: !this.state.showCorrectAnswer });
  };

  render() {
    const { showCorrectAnswer } = this.state;
    const { model, onChange, value, classes } = this.props;
    const { prompt, mode } = model;
    const modelWithValue = {
      ...model,
      value
    };
    const showCorrectAnswerToggle = mode === 'evaluate';

    return (
      <div className={classes.mainContainer}>
        {
          model.teacherInstructions && hasText(model.teacherInstructions) && (
            <React.Fragment>
              <Collapsible labels={{ hidden: 'Show Teacher Instructions', visible: 'Hide Teacher Instructions' }}>
                <div dangerouslySetInnerHTML={{ __html: model.teacherInstructions }}/>
              </Collapsible>
              <br />
            </React.Fragment>
          )
        }
        <CorrectAnswerToggle
          show={showCorrectAnswerToggle}
          toggled={showCorrectAnswer}
          onToggle={this.toggleShowCorrect}
        />
        {showCorrectAnswerToggle && <br />}
        {prompt && (
          <React.Fragment>
            <div dangerouslySetInnerHTML={{ __html: prompt }}/>
            <br />
          </React.Fragment>
        )}
        {
          model.rationale && hasText(model.rationale) && (
            <React.Fragment>
              <Collapsible labels={{ hidden: 'Show Rationale', visible: 'Hide Rationale' }}>
                <div dangerouslySetInnerHTML={{ __html: model.rationale }}/>
              </Collapsible>
              <br />
            </React.Fragment>
          )
        }
        <DraggableDragInTheBlank
          {...modelWithValue}
          onChange={onChange}
          showCorrectAnswer={showCorrectAnswer}
        />
      </div>
    );
  }
}


const styles = (theme) => ({
  mainContainer: {
    padding: theme.spacing.unit,
    color: color.text(),
    backgroundColor: color.background(),
    '& tr > td': {
      color: color.text()
    }
  },
});

export default withStyles(styles)(Main);

