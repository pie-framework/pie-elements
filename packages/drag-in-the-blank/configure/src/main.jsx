import React from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';
import EditableHtml, { ALL_PLUGINS } from '@pie-lib/editable-html';
import { InputContainer, layout, settings } from '@pie-lib/config-ui';
import { withDragContext } from '@pie-lib/drag';
import { renderMath } from '@pie-lib/math-rendering';
import { withStyles } from '@material-ui/core/styles';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import Menu from '@material-ui/core/Menu';
import MenuItem from '@material-ui/core/MenuItem';
import Typography from '@material-ui/core/Typography';

import Choices from './choices';
import { processMarkup, createSlateMarkup } from './markupUtils';
const { dropdown, toggle, Panel } = settings;

const styles = theme => ({
  promptHolder: {
    width: '100%',
    paddingBottom: theme.spacing.unit * 2,
    marginBottom: theme.spacing.unit * 2
  },
  prompt: {
    paddingTop: theme.spacing.unit * 2,
    width: '100%'
  },
  markup: {
    minHeight: '235px',
    paddingTop: theme.spacing.unit * 2,
    width: '100%',
    '& [data-slate-editor="true"]': {
      minHeight: '235px'
    }
  },
  design: {
    paddingTop: theme.spacing.unit * 3
  },
  choiceConfiguration: {
    paddingTop: theme.spacing.unit * 2,
    paddingBottom: theme.spacing.unit * 2
  },
  switchElement: {
    justifyContent: 'space-between',
    margin: 0
  },
  addButton: {
    float: 'right'
  },
  text: {
    fontFamily: 'Cerebri Sans',
    fontSize: '16px',
    lineHeight: '19px',
    color: '#495B8F'
  }
});

export class Main extends React.Component {
  static propTypes = {
    configuration: PropTypes.object.isRequired,
    model: PropTypes.object.isRequired,
    disableSidePanel: PropTypes.bool,
    onModelChanged: PropTypes.func.isRequired,
    onConfigurationChanged: PropTypes.func.isRequired,
    classes: PropTypes.object.isRequired,
    imageSupport: PropTypes.shape({
      add: PropTypes.func.isRequired,
      delete: PropTypes.func.isRequired
    })
  };

  state = {};

  componentDidUpdate() {
    // eslint-disable-next-line
    const domNode = ReactDOM.findDOMNode(this);

    renderMath(domNode);
  }

  onModelChange = newVal => {
    this.props.onModelChanged({
      ...this.props.model,
      ...newVal
    });
  };

  onPromptChanged = prompt => {
    this.props.onModelChanged({
      ...this.props.model,
      prompt
    });
  };

  onRationaleChanged = rationale => {
    this.props.onModelChanged({
      ...this.props.model,
      rationale
    });
  };

  onTeacherInstructionsChanged = teacherInstructions => {
    this.props.onModelChanged({
      ...this.props.model,
      teacherInstructions
    });
  };

  onMarkupChanged = slateMarkup => {
    this.props.onModelChanged({
      ...this.props.model,
      slateMarkup
    });
  };

  onResponsesChanged = choices => {
    const {
      model: { correctResponse, markup }
    } = this.props;
    const slateMarkup = createSlateMarkup(markup, choices, correctResponse);

    this.props.onModelChanged({
      ...this.props.model,
      slateMarkup,
      choices
    });
  };

  render() {
    const {
      classes,
      model,
      configuration,
      onConfigurationChanged,
      imageSupport
    } = this.props;
    const {
      duplicates = {},
      prompt = {},
      partialScoring = {},
      lockChoiceOrder = {},
      rationale = {},
      teacherInstructions = {},
      choicesPosition = {}
    } = configuration || {};
    const { rationaleEnabled, promptEnabled, teacherInstructionsEnabled } =
      model || {};

    return (
      <div className={classes.design}>
        <layout.ConfigLayout
          settings={
            <Panel
              model={model}
              configuration={configuration}
              onChangeModel={model => this.onModelChange(model)}
              onChangeConfiguration={configuration =>
                onConfigurationChanged(configuration, true)
              }
              groups={{
                Settings: {
                  partialScoring:
                    partialScoring.settings && toggle(partialScoring.label),
                  duplicates: duplicates.settings && toggle(duplicates.label),
                  lockChoiceOrder:
                    lockChoiceOrder.settings && toggle(lockChoiceOrder.label),
                  choicesPosition:
                    choicesPosition.settings &&
                    dropdown(choicesPosition.label, [
                      'above',
                      'below',
                      'left',
                      'right'
                    ])
                },
                Properties: {
                  teacherInstructionsEnabled:
                    teacherInstructions.settings &&
                    toggle(teacherInstructions.label),
                  rationaleEnabled:
                    rationale.settings && toggle(rationale.label),
                  promptEnabled: prompt.settings && toggle(prompt.label)
                }
              }}
            />
          }
        >
          <div>
            {teacherInstructionsEnabled && (
              <InputContainer
                label={teacherInstructions.label}
                className={classes.promptHolder}
              >
                <EditableHtml
                  className={classes.prompt}
                  markup={model.teacherInstructions || ''}
                  onChange={this.onTeacherInstructionsChanged}
                  imageSupport={imageSupport}
                  nonEmpty={false}
                />
              </InputContainer>
            )}
            {promptEnabled && (
              <InputContainer
                label={prompt.label}
                className={classes.promptHolder}
              >
                <EditableHtml
                  className={classes.prompt}
                  markup={model.prompt}
                  onChange={this.onPromptChanged}
                  imageSupport={imageSupport}
                  nonEmpty={!prompt.settings}
                  disableUnderline
                />
              </InputContainer>
            )}
            <Typography className={classes.text}>
              Define Template, Choices, and Correct Responses
            </Typography>
            <EditableHtml
              activePlugins={ALL_PLUGINS}
              toolbarOpts={{
                position: 'top',
                alwaysVisible: true
              }}
              responseAreaProps={{
                type: 'drag-in-the-blank',
                options: {
                  duplicates: model.duplicates
                }
              }}
              className={classes.markup}
              markup={model.slateMarkup}
              onChange={this.onMarkupChanged}
              imageSupport={imageSupport}
              nonEmpty={!prompt.settings}
              disableUnderline
            />
            <Choices
              model={model}
              duplicates={model.duplicates}
              onChange={this.onResponsesChanged}
            />
            {rationaleEnabled && (
              <InputContainer
                label={rationale.label}
                className={classes.promptHolder}
              >
                <EditableHtml
                  className={classes.prompt}
                  markup={model.rationale || ''}
                  onChange={this.onRationaleChanged}
                  imageSupport={imageSupport}
                />
              </InputContainer>
            )}
          </div>
        </layout.ConfigLayout>
      </div>
    );
  }
}

const Styled = withStyles(styles)(Main);

export default withDragContext(Styled);
