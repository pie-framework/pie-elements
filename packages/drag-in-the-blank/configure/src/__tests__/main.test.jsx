import { shallow } from 'enzyme';
import React from 'react';

import { Main } from '../main';
import sensibleDefaults from "../defaults";
import { createSlateMarkup, processMarkup } from "../markupUtils";

jest.mock('@pie-lib/config-ui', () => ({
  choiceUtils: {
    firstAvailableIndex: jest.fn()
  },
  settings: {
    Panel: props => <div {...props} />,
    toggle: jest.fn(),
    radio: jest.fn(),
    dropdown: jest.fn()
  },
  layout: {
    ConfigLayout: props => <div>{props.children}</div>
  },
  InputContainer: props => <div>{props.children}</div>
}));

const model = {
  markup: '{{0}} + {{1}} = 15',
  prompt: '<p>Solve the equation below.</p>',
  choices: [
    { value: '<div>6</div>', id: '0' },
    { value: '<div>9</div>', id: '1' }
  ],
  choicesPosition: 'below',
  correctResponse: {
    '0': '0',
    '1': '1'
  },
  duplicates: true,
  alternateResponses: [
    ['1'],
    ['0']
  ],
  rationale: '<p>A correct response is shown below:</p><ul><li>2/6 = 1/3</li><li>4/8 = 1/2</li><li>6/10 = 3/5</li><li>9/12 = 3/4</li></ul>'
};

const prepareModel = (model = {}) => {
  const joinedObj = {
    ...sensibleDefaults.model,
    ...model
  };
  const slateMarkup = model.slateMarkup ||
    createSlateMarkup(joinedObj.markup, joinedObj.choices, joinedObj.correctResponse);
  const processedMarkup = processMarkup(slateMarkup);

  return {
    ...joinedObj,
    slateMarkup,
    markup: processedMarkup.markup,
    correctResponse: processedMarkup.correctResponse
  };
};

describe('Main', () => {
  let onModelChanged = jest.fn();
  let onConfigurationChanged = jest.fn();

  const wrapper = extras => {
    const defaults = {
      onModelChanged,
      onConfigurationChanged,
      classes: {},
      model: prepareModel({
        ...model,
        ...extras
      }),
      configuration: sensibleDefaults.configuration
    };
    const props = { ...defaults };

    return shallow(<Main {...props} />);
  };

  describe('snapshot', () => {
    it('renders with teacher instructions, prompt and rationale even if not set', () => {
      expect(wrapper()).toMatchSnapshot();
    });

    it('renders without teacher instructions, prompt and rationale', () => {
      expect(wrapper({
        promptEnabled: false,
        teacherInstructionsEnabled: false,
        rationaleEnabled: false
      })).toMatchSnapshot();
    });
  });

  describe('logic', () => {
    let w;

    beforeEach(() => {
      w = wrapper();
    });

    describe('onModelChange', () => {
      it('changes the model', () => {
        w.instance().onModelChange({ promptEnabled: false });

        expect(onModelChanged).toBeCalledWith({
          ...prepareModel(model),
          promptEnabled: false
        });
      });
    });

    describe('onPromptChanged', () => {
      it('changes the prompt value', () => {
        w.instance().onPromptChanged('This is the new prompt');

        expect(onModelChanged).toBeCalledWith({
          ...prepareModel(model),
          prompt: 'This is the new prompt'
        });
      });
    });

    describe('onRationaleChanged', () => {
      it('changes the rationale value', () => {
        w.instance().onRationaleChanged('New Rationale');

        expect(onModelChanged).toBeCalledWith({
          ...prepareModel(model),
          rationale: 'New Rationale'
        });
      });
    });

    describe('onTeacherInstructionsChanged', () => {
      it('changes the teacher instructions value', () => {
        w.instance().onTeacherInstructionsChanged('New Teacher Instructions');

        expect(onModelChanged).toBeCalledWith({
          ...prepareModel(model),
          teacherInstructions: 'New Teacher Instructions'
        });
      });
    });

    describe('onMarkupChanged', () => {
      it('changes slate markup value', () => {
        const slateMarkup = '<span data-type="drag_in_the_blank" data-index="0" data-id="0" data-value="&lt;div&gt;6&lt;/div&gt;"></span> + <span data-type="drag_in_the_blank" data-index="1" data-id="1" data-value="&lt;div&gt;9&lt;/div&gt;"></span> = 15eggs';

        w.instance().onMarkupChanged(slateMarkup);

        expect(onModelChanged).toBeCalledWith({
          ...prepareModel(model),
          slateMarkup
        });
      });
    });

    describe('onResponsesChanged', () => {
      it('changes choices and slateMarkup as well', () => {
        const newChoices = [
          { value: '<div>6</div>', id: '0' },
          { value: '<div>3^2</div>', id: '1' }
        ];

        w.instance().onResponsesChanged(newChoices);

        expect(onModelChanged).toBeCalledWith({
          ...prepareModel(model),
          slateMarkup: '<span data-type="drag_in_the_blank" data-index="0" data-id="0" data-value="&lt;div&gt;6&lt;/div&gt;"></span> + <span data-type="drag_in_the_blank" data-index="1" data-id="1" data-value="&lt;div&gt;3^2&lt;/div&gt;"></span> = 15',
          choices: newChoices
        });
      });
    });
  });
});
