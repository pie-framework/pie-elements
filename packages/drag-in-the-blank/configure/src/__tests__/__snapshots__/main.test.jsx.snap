// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main snapshot renders with teacher instructions, prompt and rationale even if not set 1`] = `
<div>
  <ConfigLayout
    settings={
      <Panel
        configuration={
          Object {
            "choicesPosition": Object {
              "label": "Choices Position",
              "settings": true,
            },
            "duplicates": Object {
              "label": "Duplicates",
              "settings": true,
            },
            "lockChoiceOrder": Object {
              "label": "Lock Choice Order",
              "settings": true,
            },
            "partialScoring": Object {
              "label": "Allow Partial Scoring",
              "settings": false,
            },
            "prompt": Object {
              "label": "Prompt",
              "settings": true,
            },
            "rationale": Object {
              "label": "Rationale",
              "settings": true,
            },
            "teacherInstructions": Object {
              "label": "Teacher Instructions",
              "settings": true,
            },
          }
        }
        groups={
          Object {
            "Properties": Object {
              "promptEnabled": undefined,
              "rationaleEnabled": undefined,
              "teacherInstructionsEnabled": undefined,
            },
            "Settings": Object {
              "choicesPosition": undefined,
              "duplicates": undefined,
              "lockChoiceOrder": undefined,
              "partialScoring": false,
            },
          }
        }
        model={
          Object {
            "alternateResponses": Array [
              Array [
                "1",
              ],
              Array [
                "0",
              ],
            ],
            "choices": Array [
              Object {
                "id": "0",
                "value": "<div>6</div>",
              },
              Object {
                "id": "1",
                "value": "<div>9</div>",
              },
            ],
            "choicesPosition": "below",
            "correctResponse": Object {
              "0": "0",
              "1": "1",
            },
            "disabled": false,
            "duplicates": true,
            "markup": "{{0}} + {{1}} = 15",
            "mode": "gather",
            "prompt": "<p>Solve the equation below.</p>",
            "promptEnabled": true,
            "rationale": "<p>A correct response is shown below:</p><ul><li>2/6 = 1/3</li><li>4/8 = 1/2</li><li>6/10 = 3/5</li><li>9/12 = 3/4</li></ul>",
            "rationaleEnabled": true,
            "shuffle": true,
            "slateMarkup": "<span data-type=\\"drag_in_the_blank\\" data-index=\\"0\\" data-id=\\"0\\" data-value=\\"&lt;div&gt;6&lt;/div&gt;\\"></span> + <span data-type=\\"drag_in_the_blank\\" data-index=\\"1\\" data-id=\\"1\\" data-value=\\"&lt;div&gt;9&lt;/div&gt;\\"></span> = 15",
            "studentInstructionsEnabled": true,
            "teacherInstructionsEnabled": true,
          }
        }
        onChangeConfiguration={[Function]}
        onChangeModel={[Function]}
      />
    }
  >
    <div>
      <InputContainer
        label="Teacher Instructions"
      >
        <EditableHtml
          markup=""
          nonEmpty={false}
          onChange={[Function]}
        />
      </InputContainer>
      <InputContainer
        label="Prompt"
      >
        <EditableHtml
          disableUnderline={true}
          markup="<p>Solve the equation below.</p>"
          nonEmpty={false}
          onChange={[Function]}
        />
      </InputContainer>
      <WithStyles(Typography)>
        Define Template, Choices, and Correct Responses
      </WithStyles(Typography)>
      <EditableHtml
        disableUnderline={true}
        markup="<span data-type=\\"drag_in_the_blank\\" data-index=\\"0\\" data-id=\\"0\\" data-value=\\"&lt;div&gt;6&lt;/div&gt;\\"></span> + <span data-type=\\"drag_in_the_blank\\" data-index=\\"1\\" data-id=\\"1\\" data-value=\\"&lt;div&gt;9&lt;/div&gt;\\"></span> = 15"
        nonEmpty={false}
        onChange={[Function]}
        pluginProps={
          Object {
            "audio": Object {
              "disabled": true,
            },
            "video": Object {
              "disabled": true,
            },
          }
        }
        responseAreaProps={
          Object {
            "options": Object {
              "duplicates": true,
            },
            "type": "drag-in-the-blank",
          }
        }
        toolbarOpts={
          Object {
            "position": "top",
          }
        }
      />
      <WithStyles(Choices)
        duplicates={true}
        model={
          Object {
            "alternateResponses": Array [
              Array [
                "1",
              ],
              Array [
                "0",
              ],
            ],
            "choices": Array [
              Object {
                "id": "0",
                "value": "<div>6</div>",
              },
              Object {
                "id": "1",
                "value": "<div>9</div>",
              },
            ],
            "choicesPosition": "below",
            "correctResponse": Object {
              "0": "0",
              "1": "1",
            },
            "disabled": false,
            "duplicates": true,
            "markup": "{{0}} + {{1}} = 15",
            "mode": "gather",
            "prompt": "<p>Solve the equation below.</p>",
            "promptEnabled": true,
            "rationale": "<p>A correct response is shown below:</p><ul><li>2/6 = 1/3</li><li>4/8 = 1/2</li><li>6/10 = 3/5</li><li>9/12 = 3/4</li></ul>",
            "rationaleEnabled": true,
            "shuffle": true,
            "slateMarkup": "<span data-type=\\"drag_in_the_blank\\" data-index=\\"0\\" data-id=\\"0\\" data-value=\\"&lt;div&gt;6&lt;/div&gt;\\"></span> + <span data-type=\\"drag_in_the_blank\\" data-index=\\"1\\" data-id=\\"1\\" data-value=\\"&lt;div&gt;9&lt;/div&gt;\\"></span> = 15",
            "studentInstructionsEnabled": true,
            "teacherInstructionsEnabled": true,
          }
        }
        onChange={[Function]}
      />
      <InputContainer
        label="Rationale"
      >
        <EditableHtml
          markup="<p>A correct response is shown below:</p><ul><li>2/6 = 1/3</li><li>4/8 = 1/2</li><li>6/10 = 3/5</li><li>9/12 = 3/4</li></ul>"
          onChange={[Function]}
        />
      </InputContainer>
    </div>
  </ConfigLayout>
</div>
`;

exports[`Main snapshot renders without teacher instructions, prompt and rationale 1`] = `
<div>
  <ConfigLayout
    settings={
      <Panel
        configuration={
          Object {
            "choicesPosition": Object {
              "label": "Choices Position",
              "settings": true,
            },
            "duplicates": Object {
              "label": "Duplicates",
              "settings": true,
            },
            "lockChoiceOrder": Object {
              "label": "Lock Choice Order",
              "settings": true,
            },
            "partialScoring": Object {
              "label": "Allow Partial Scoring",
              "settings": false,
            },
            "prompt": Object {
              "label": "Prompt",
              "settings": true,
            },
            "rationale": Object {
              "label": "Rationale",
              "settings": true,
            },
            "teacherInstructions": Object {
              "label": "Teacher Instructions",
              "settings": true,
            },
          }
        }
        groups={
          Object {
            "Properties": Object {
              "promptEnabled": undefined,
              "rationaleEnabled": undefined,
              "teacherInstructionsEnabled": undefined,
            },
            "Settings": Object {
              "choicesPosition": undefined,
              "duplicates": undefined,
              "lockChoiceOrder": undefined,
              "partialScoring": false,
            },
          }
        }
        model={
          Object {
            "alternateResponses": Array [
              Array [
                "1",
              ],
              Array [
                "0",
              ],
            ],
            "choices": Array [
              Object {
                "id": "0",
                "value": "<div>6</div>",
              },
              Object {
                "id": "1",
                "value": "<div>9</div>",
              },
            ],
            "choicesPosition": "below",
            "correctResponse": Object {
              "0": "0",
              "1": "1",
            },
            "disabled": false,
            "duplicates": true,
            "markup": "{{0}} + {{1}} = 15",
            "mode": "gather",
            "prompt": "<p>Solve the equation below.</p>",
            "promptEnabled": false,
            "rationale": "<p>A correct response is shown below:</p><ul><li>2/6 = 1/3</li><li>4/8 = 1/2</li><li>6/10 = 3/5</li><li>9/12 = 3/4</li></ul>",
            "rationaleEnabled": false,
            "shuffle": true,
            "slateMarkup": "<span data-type=\\"drag_in_the_blank\\" data-index=\\"0\\" data-id=\\"0\\" data-value=\\"&lt;div&gt;6&lt;/div&gt;\\"></span> + <span data-type=\\"drag_in_the_blank\\" data-index=\\"1\\" data-id=\\"1\\" data-value=\\"&lt;div&gt;9&lt;/div&gt;\\"></span> = 15",
            "studentInstructionsEnabled": true,
            "teacherInstructionsEnabled": false,
          }
        }
        onChangeConfiguration={[Function]}
        onChangeModel={[Function]}
      />
    }
  >
    <div>
      <WithStyles(Typography)>
        Define Template, Choices, and Correct Responses
      </WithStyles(Typography)>
      <EditableHtml
        disableUnderline={true}
        markup="<span data-type=\\"drag_in_the_blank\\" data-index=\\"0\\" data-id=\\"0\\" data-value=\\"&lt;div&gt;6&lt;/div&gt;\\"></span> + <span data-type=\\"drag_in_the_blank\\" data-index=\\"1\\" data-id=\\"1\\" data-value=\\"&lt;div&gt;9&lt;/div&gt;\\"></span> = 15"
        nonEmpty={false}
        onChange={[Function]}
        pluginProps={
          Object {
            "audio": Object {
              "disabled": true,
            },
            "video": Object {
              "disabled": true,
            },
          }
        }
        responseAreaProps={
          Object {
            "options": Object {
              "duplicates": true,
            },
            "type": "drag-in-the-blank",
          }
        }
        toolbarOpts={
          Object {
            "position": "top",
          }
        }
      />
      <WithStyles(Choices)
        duplicates={true}
        model={
          Object {
            "alternateResponses": Array [
              Array [
                "1",
              ],
              Array [
                "0",
              ],
            ],
            "choices": Array [
              Object {
                "id": "0",
                "value": "<div>6</div>",
              },
              Object {
                "id": "1",
                "value": "<div>9</div>",
              },
            ],
            "choicesPosition": "below",
            "correctResponse": Object {
              "0": "0",
              "1": "1",
            },
            "disabled": false,
            "duplicates": true,
            "markup": "{{0}} + {{1}} = 15",
            "mode": "gather",
            "prompt": "<p>Solve the equation below.</p>",
            "promptEnabled": false,
            "rationale": "<p>A correct response is shown below:</p><ul><li>2/6 = 1/3</li><li>4/8 = 1/2</li><li>6/10 = 3/5</li><li>9/12 = 3/4</li></ul>",
            "rationaleEnabled": false,
            "shuffle": true,
            "slateMarkup": "<span data-type=\\"drag_in_the_blank\\" data-index=\\"0\\" data-id=\\"0\\" data-value=\\"&lt;div&gt;6&lt;/div&gt;\\"></span> + <span data-type=\\"drag_in_the_blank\\" data-index=\\"1\\" data-id=\\"1\\" data-value=\\"&lt;div&gt;9&lt;/div&gt;\\"></span> = 15",
            "studentInstructionsEnabled": true,
            "teacherInstructionsEnabled": false,
          }
        }
        onChange={[Function]}
      />
    </div>
  </ConfigLayout>
</div>
`;
